---
title: "PhyloView"
output: html_document
params:
  clade: formicidae
---


```{r, include=FALSE, echo=FALSE}

if(!require("devtools")){
    install.packages("devtools")
    library(devtools)
}
if(!require("rphylotastic")){
    devtools::install_github("phylotastic/rphylotastic")
    library(rphylotastic)
}
if(!require("rotl")){
    install.packages("rotl")
    library(rotl)
}
if(!require("ape")){
    install.packages("ape")
    library(ape)
}
if(!require("rentrez")){
    install.packages("rentrez")
    library(rentrez)
}
if(!require("knitr")){
    install.packages("knitr")
    library(knitr)
}
```

```{r, include=FALSE, echo=FALSE}
clade.name <- rphylotastic::ResolveNamesWithOToL(params$clade)
id <- ott_id(tnrs_match_names(clade.name))
node.info <- tol_node_info(id)
relevant.studies <- studies_find_trees(property="ot:ottTaxonName", value=clade.name)
tree.info <- data.frame()
all.trees <- list_trees(relevant.studies)
for (study.index in sequence(nrow(relevant.studies))) {
  for (tree.index in sequence(length(all.trees[study.index]))) {
    phy <- get_study_tree(study_id=relevant.studies$study_ids[study.index], tree_id = all.trees[[study.index]][tree.index])
    local.result <- data.frame(Year=relevant.studies$study_year[study.index], Ntax=Ntip(phy), Title=relevant.studies$title[study.index], DOI=relevant.studies$study_doi[study.index])
    if(nrow(tree.info)==0) {
      tree.info <- local.result
    } else {
      tree.info <- rbind(tree.info, local.result)
    }
  }
}
```

```{r, include=FALSE, echo=FALSE}
genbank.species.query <- paste0(clade.name, '[subtree] AND species[rank] AND specified[prop]')
genbank.species.count <-  entrez_search(db="taxonomy", genbank.species.query, use_history=TRUE)$count
```


Taxon **`r clade.name`** has `r node.info$num_tips` terminal taxa in OpenTree. This is based on taxonomy plus `r nrow(relevant.studies)` studies.


```{r, echo=FALSE}
kable(tree.info, caption="Supporting studies in OpenTree (if any)")
```

In GenBank, there are `r genbank.species.count` species listed for this group.

```{r, echo=FALSE}
focal.genes <- c("COI", "18S", "28S", "matk", "rbcl")
GetNucCount <- function(gene, taxon=clade.name) {
  gene.query <- paste0(taxon, '[organism] AND ',gene)
  Sys.sleep(3) #just to make sure we stay nice
  return(entrez_search(db="nuccore", gene.query, use_history=TRUE)$count)
}
gene.count <- sapply(focal.genes, GetNucCount, taxon=clade.name) #make sure not to do mclapply or similar lest you violate NCBI terms of service
#print(gene.count)
kable(data.frame(seqs=gene.count), caption="Number of sequences in GenBank")
```

```{r, echo=FALSE}
pubmed.query <- paste0(clade.name, ' AND phylogeny')
pubmed.result <- entrez_search(db="pubmed", pubmed.query, use_history=TRUE)
pubmed.summaries <- entrez_summary(db="pubmed", web_history=pubmed.result$web_history)
pubmed.df <- data.frame(Date=extract_from_esummary(pubmed.summaries, elements=c("sortpubdate")), FirstAuthor=extract_from_esummary(pubmed.summaries, elements=c("sortfirstauthor")), Journal=extract_from_esummary(pubmed.summaries, elements=c("fulljournalname")), Title=extract_from_esummary(pubmed.summaries, elements=c("title")))
pubmed.df <- pubmed.df[order(pubmed.df$Date, decreasing=TRUE),]
kable(pubmed.df, caption="Publications")
```
